src/cpp/src/logit_processor.hpp

```diff
@@ -16,12 +16,35 @@ struct Token {
     Token() = default;
 };
 
+struct Logits {
+    float * m_data = nullptr;
+    size_t m_size;
+    // Late initialized
+    std::vector<Token> m_vector;
+
+    Logits(float* data, size_t size): m_data(data), m_size(size) {}
+
+
+    void initialize_vector() {
+        OPENVINO_ASSERT(m_vector.size() == 0, "Logits vector already initialized");
+        m_vector.reserve(m_size);
```

Comment:
Are there use cases when `initialize_vector()` isn't called. If there are none, you can avoid storing `m_data` and move `m_vector` filling to the constructor.

Reply:
Yes, `initialize_vector` is called only if `top_p` or `top_k` are specified (when we need sorting). So for example if only `temperature` and some penalties are defined, we use `m_data` and `m_size` and `m_vector` is unused.